### PR TITLE
Add retry to juju status.

### PR DESCRIPTION
**Pull request by howbazaar**
Add a default retry inside 'juju status'.

The juju-wait plugin calls 'juju status' repeatedly. The solutions QA setup has some proxy in the middle and the client through the server was disconnecting and the server throught the client was disconnecting. This results in bung runs. This fix is primarily targetted to fix this issue.

## Documentation changes

Internal implementation change only. Nothing to really change here.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1782803

### CODE REVIEW

File: cmd/juju/status/export_test.go [hunk 1]
```diff
@@ -3,8 +3,16 @@
 
 package status
 
-import "github.com/juju/cmd"
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
```
Comment (anastasiamac) [inline cmd/juju/status/export_test.go]: organise me

File: cmd/juju/status/status.go [hunk 2]
```diff
@@ -119,6 +130,9 @@ func (c *statusCommand) SetFlags(f *gnuflag.FlagSet) {
 
 	f.BoolVar(&c.relations, "relations", false, "Show 'relations' section")
 
+	f.IntVar(&c.retryCount, "retry-count", 3, "Number of times to retry API failures.")
```
Comment (anastasiamac) [inline cmd/juju/status/status.go]: No fullstops at the end of a sentence :D

File: cmd/juju/status/status_internal_test.go [hunk 3]
```diff
@@ -0,0 +1,5475 @@
+// Copyright 2015 Canonical Ltd.
```
Comment (anastasiamac) [inline cmd/juju/status/status_internal_test.go]: i wish it was still 2015 :)

Comment (howbazaar) [reply cmd/juju/status/status_internal_test.go]: me too

File: cmd/juju/status/export_test.go (hunk 1 above)
Comment (howbazaar) [reply cmd/juju/status/export_test.go]: done

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**